forge/src/main/java/whocraft/tardis_refined/common/data/DesktopProvider.java

```diff
@@ -0,0 +1,74 @@
+package whocraft.tardis_refined.common.data;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import net.minecraft.ChatFormatting;
+import net.minecraft.data.CachedOutput;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.DataProvider;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.Style;
+import net.minecraft.resources.ResourceLocation;
+import whocraft.tardis_refined.TardisRefined;
+import whocraft.tardis_refined.common.tardis.TardisDesktops;
+import whocraft.tardis_refined.common.tardis.themes.DesktopTheme;
+import whocraft.tardis_refined.common.util.MiscHelper;
+
+import java.io.IOException;
+
+import static whocraft.tardis_refined.common.tardis.TardisDesktops.DESKTOPS;
+import static whocraft.tardis_refined.common.tardis.TardisDesktops.addDesktop;
+
+public class DesktopProvider implements DataProvider {
+
+    protected final DataGenerator generator;
+
+
+    public DesktopProvider(DataGenerator generator) {
+        Preconditions.checkNotNull(generator);
+        this.generator = generator;
+    }
+
+    public static void registerDesktops() {
+
+        addDesktop(TardisDesktops.DEFAULT_OVERGROWN_THEME);
+        addDesktop(TardisDesktops.FACTORY_THEME);
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "coral"), new ResourceLocation(TardisRefined.MODID, "desktop/coral"), true));
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "toyota"), new ResourceLocation(TardisRefined.MODID, "desktop/toyota"), true));
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "copper"), new ResourceLocation(TardisRefined.MODID, "desktop/copper"), true));
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "crystal"), new ResourceLocation(TardisRefined.MODID, "desktop/crystal"), true));
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "victorian"), new ResourceLocation(TardisRefined.MODID, "desktop/victorian"), true));
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "nuka"), new ResourceLocation(TardisRefined.MODID, "desktop/nuka"), true));
+        addDesktop(new DesktopTheme(new ResourceLocation(TardisRefined.MODID, "future_nostalgia"), new ResourceLocation(TardisRefined.MODID, "desktop/future_nostalgia"), true));
```

Comment:
I feel like adding a Tardis Refined specific constructor would reduce the complexity of the datagen code.

Reply:
Good idea.